js_modules/dagster-ui/packages/ui-core/src/pipelines/JobTabs.tsx

```diff
@@ -0,0 +1,117 @@
+import {Tab, Tabs, Tooltip} from '@dagster-io/ui-components';
+import * as React from 'react';
+
+import {PermissionResult, PermissionsState, permissionResultForKey} from '../app/Permissions';
+import {TabLink} from '../ui/TabLink';
+import {RepoAddress} from '../workspace/types';
+import {workspacePathFromAddress} from '../workspace/workspacePath';
+
+import {ExplorerPath, explorerPathToString} from './PipelinePathUtils';
+
+export const DEFAULT_JOB_TAB_ORDER = ['overview', 'playground', 'runs', 'partitions'];
+
+interface Props {
+  repoAddress: RepoAddress;
+  isJob: boolean;
+  explorerPath: ExplorerPath;
+  matchingTab?: string;
+  permissions: PermissionsState;
+  tabs: JobTabConfig[];
+}
+
+export const JobTabs = (props: Props) => {
+  const {repoAddress, isJob, explorerPath, matchingTab = '', permissions, tabs} = props;
+
+  const explorerPathForTab = explorerPathToString({
+    ...explorerPath,
+    opNames: [],
+  });
+
+  const selectedTab = React.useMemo(() => {
+    return (
+      tabs.find((tab) => tab.pathComponent === matchingTab) ||
+      tabs.find((tab) => tab.pathComponent === '')
+    );
+  }, [matchingTab, tabs]);
+
+  return (
+    <Tabs size="large" selectedTabId={selectedTab!.id}>
+      {tabs
+        .filter((tab) => !tab.isHidden)
+        .map((tab) => {
+          const {id, title: text, getPermissionsResult} = tab;
+          const permissionsResult = getPermissionsResult ? getPermissionsResult(permissions) : null;
+          const disabled = !!(permissionsResult && !permissionsResult.enabled);
+          const title =
+            permissionsResult && disabled ? (
+              <Tooltip content={permissionsResult.disabledReason} placement="top">
+                {text}
+              </Tooltip>
+            ) : (
+              text
+            );
+
+          const href = workspacePathFromAddress(
+            repoAddress,
+            `/${isJob ? 'jobs' : 'pipelines'}/${explorerPathForTab}${tab.pathComponent}`,
```

Comment:
It seems like these three lines are the only parts of this that are really job-specific -- I wonder if we could make this a top level "DynamicTabs" component and pass this bits in as a `basePath` 

May not be worth the abstraction, but I think this closely mirrors / adds features to `AssetTabs.tsx`  and that could probably use this version!

Reply:
One of the other key differences is that the Job tabs use a subpath to determine the selected tab, while Asset tabs use a querystring.